cmake/asan.cmake

```diff
@@ -0,0 +1,26 @@
+get_property(isMultiConfig GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG)
+
+if(isMultiConfig)
+    if(NOT "ASan" IN_LIST CMAKE_CONFIGURATION_TYPES)
+        list(APPEND CMAKE_CONFIGURATION_TYPES Asan)
+    endif()
+else()
+    set(allowedBuildTypes ASan Debug Release RelWithDebInfo MinSizeRel)
+    set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "${allowedBuildTypes}")
+endif()
```

Comment:
Still don't see the world where this is nicer, simpler and easier to understand than autotools...

Reply:
How is this helpful?